File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.go
func firstUniqChar(s string) int {
	var counts = make([]int, 26)
	
	for i := 0; i < len(s); i++ {
		counts[s[i] - 'a']++
	}


    for i := 0; i < len(s); i++ {
		if counts[s[i] - 'a'] == 1 {
			return i
		}
	}

	return -1
}

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
var firstUniqChar = function(s) {
    let result =-1;
    let hash = {}
    
    for(let i=0; i<s.length; i++) {
        if(!hash[s[i]]) hash[s[i]] = [i]
        else hash[s[i]].push(i)
    }
    
    for(const [key, values] of Object.entries(hash)){
        if(values.length === 1) return values[0]
    }
    
    return result;
};